mlkem/asm/clean/rej_uniform_asm.s

```diff
@@ -0,0 +1,369 @@
+    lsr     x8, x8, #37
+    add     w8, w8, w8, lsl #1
+    lsl     \dst, w8, #4
+.endm
+
+.align 4
+.global     rej_uniform_asm
+.global     _rej_uniform_asm
+rej_uniform_asm:
+_rej_uniform_asm:
+
+    /* Input registers */
+    stack                       .req x0
+    buf                         .req x1
+    buf_consumed                .req x2
+    buflen                      .req w3
+    len                         .req x4
+    lenw                        .req w4
+    table_idx                   .req x5
+    bit_table                   .req x6
+
+    /* Output registers */
+    output                      .req x11
+    sp_copy                     .req x16
+    buf_consumed_cpy            .req x17
+
+    /* Temporary registers */
+    tmp                         .req w8
+    count                       .req w8
+    bound                       .req w9
+    minw                        .req w8
+    min                         .req x8
+    iterw                       .req w10
+    iter                        .req x10
+
+    rec_idx_0                   .req w12
+    rec_idx_1                   .req w13
+    rec_idx_2                   .req w14
+    rec_idx_3                   .req w15
+
+    ctr0                        .req w12
+    ctr1                        .req w13
+    ctr2                        .req w14
+    ctr3                        .req w15
+
+    /* Vector registers */
+
+    buf0                        .req v0
+    buf1                        .req v1
+    buf2                        .req v2
+
+    tmp0                        .req v4
+    tmp1                        .req v5
+    tmp2                        .req v6
+    tmp3                        .req v7
+
+    sign0                       .req v4
+    sign1                       .req v5
+    sign2                       .req v6
+    sign3                       .req v7
+
+    val0                        .req v16
+    val0q                       .req q16
+    val1                        .req v17
+    val1q                       .req q17
+    val2                        .req v18
+    val2q                       .req q18
+    val3                        .req v19
+    val3q                       .req q19
+
+    t0                          .req s20
+    t1                          .req s21
+    t2                          .req s22
+    t3                          .req s23
+
+    table0                      .req v24
+    table0q                     .req q24
+    table1                      .req v25
+    table1q                     .req q25
+    table2                      .req v26
+    table2q                     .req q26
+    table3                      .req v27
+    table3q                     .req q27
+
+    kyber_q                     .req v30
+    bits                        .req v31
+    bits_q                      .req q31
+
+
+    /* 
+     * Preserve registers and restore at the end of function.
+     * Stack alignment must be 16 bytes. 
+     */
+    sub     sp, sp, #0x270
```

Comment:
Can you leave a comment describing the usage and layout of the stack?

Reply:
Sure, will do now.

Reply:
added in https://github.com/pq-code-package/mlkem-c-aarch64/pull/47/commits/908164c2534b4d6dc24e43cd1ad8cd6fe4ffcbd2